### PR TITLE
[#156101143][#156213326] Feature chat mentions upgrade

### PR DESCRIPTION
**Pull request by armandgray**
1. [#156101143] after deleting an @ mention moving the cursor to right after the “@” symbol
2. [#156213326] sending on enter when clicked anywhere in the message

### CODE REVIEW

File: src/MentionsTextInput.js [hunk 1]
```diff
@@ -371,6 +370,31 @@ export default class MentionsTextInput extends Component {
     }
   }
 
+  setSelection(start, end = start) {
```
Comment (leoleblanc) [inline src/MentionsTextInput.js:373]: what are `start` and `end`?

Comment (armandgray) [reply src/MentionsTextInput.js:373]: the start and end of the selection!

Comment (armandgray) [reply src/MentionsTextInput.js:373]: eg: 1234 -> you could select the chars from 0, 4

File: src/MentionsTextInput.js [hunk 2]
```diff
@@ -526,12 +569,17 @@ export default class MentionsTextInput extends Component {
     return !text1 && text2.length > 1 || text1.length < text2.length - 1;
   }
 
+  hasNewLineChar(text) {
+    return text && text.length > 0 && text[text.length - 1] == '\n'
+              || text.indexOf('\n') !== -1;
```
Comment (leoleblanc) [inline src/MentionsTextInput.js]: These two checks at the end seem redundant, it looks like you could get rid of the `text[text.length - 1] == '\n'`?

File: src/MentionsTextInput.js (hunk 1 above)
Comment (leoleblanc) [reply src/MentionsTextInput.js:373]: oh so these are indices, that wasn't too clear, but was able to gather from context.  Might be useful to use flowtypes to say that they're numbers, e.g. `start: number, end: number=start`

Comment (leoleblanc) [reply src/MentionsTextInput.js:373]: Or maybe call it `setSelectionFromIndices`?

Comment (armandgray) [reply src/MentionsTextInput.js:373]: its called selection its a prop

Comment (armandgray) [reply src/MentionsTextInput.js:373]: the name is consistent with the name of the prop and the name of the passed param to `onSelectionChange`

Comment (armandgray) [reply src/MentionsTextInput.js:373]: ie `selection: {start, end}` is what RN has set as the naming convention

Comment (leoleblanc) [reply src/MentionsTextInput.js:373]: gotcha

**Review by leoleblanc — COMMENTED** [review]
👍

**Review by MattFoley — APPROVED** [review]
👍

**Pull request merged**